scripts/publish.sh

```diff
@@ -351,6 +351,19 @@ verify_ecr() {
 
 check_image_version() {
 	export DOCKER_CLI_EXPERIMENTAL=enabled
+
+	username="$(aws secretsmanager get-secret-value --secret-id $DOCKER_HUB_SECRET --region us-west-2 | jq -r '.SecretString | fromjson.username')"
+	password="$(aws secretsmanager get-secret-value --secret-id $DOCKER_HUB_SECRET --region us-west-2 | jq -r '.SecretString | fromjson.password')"
+
+	# Logout when the script exits
+	trap cleanup EXIT
+	cleanup() {
+		docker logout
+	}
+
+	# login to DockerHub
+	docker login -u "${username}" --password "${password}"
+	
```

Comment:
We are doing this in at least two different places. Is it possible to extract them in a function to avoid duplicate code?

Reply:
good point. We can do that to make the code more concise

Reply:
+1